clients/ts/signalr/tests/ServerSentEventsTransport.test.ts

```diff
@@ -0,0 +1,177 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+import { TransferFormat } from "../src/ITransport";
+
+import { HttpClient, HttpRequest } from "../src/HttpClient";
+import { ILogger } from "../src/ILogger";
+import { ServerSentEventsTransport } from "../src/ServerSentEventsTransport";
+import { VerifyLogger } from "./Common";
+import { TestEventSource, TestMessageEvent } from "./TestEventSource";
+import { TestHttpClient } from "./TestHttpClient";
+
+describe("ServerSentEventsTransport", () => {
+    it("does not allow non-text formats", async () => {
+        await VerifyLogger.run(async (logger) => {
+            const sse = new ServerSentEventsTransport(new TestHttpClient(), undefined, logger, true, TestEventSource);
+
+            await expect(sse.connect("", TransferFormat.Binary))
+                .rejects
+                .toMatchObject({ message: "The Server-Sent Events transport only supports the 'Text' transfer format" });
```

Comment:
I think `.toEqual(new Error("..."))` will work; according to a quick test I did in node:

```
> expect(new Error("foo")).toEqual(new Error("bar"))
Error: expect(received).toEqual(expected)

Expected value to equal:
  [Error: bar]
Received:
  [Error: foo]

Difference:

- Expected
+ Received

- [Error: bar]
+ [Error: foo]
> expect(new Error("foo")).toEqual(new Error("foo"))
undefined
```